### ISSUE TITLE
Questions on  _fdataloader

### ISSUE THREAD

**Issue by Eie1** [Rikorose/DeepFilterNet#180]
could I ask that is there any code can be refered to replace this part : from libdfdata import _FdDataLoader
  in DeepFilterNet-main/pyDF-data/libdfdata/torch_dataloader.py? And is it right that only by replacing this line code then I can transform the code of dataloader into a pure python format？If I could only refer the code in  DeepFilterNet-main/pyDF-data/src/lib.rs？

Thanks for helping very much

**Comment by Rikorose** [comment]
No you would need to change a lot more. As I said previously the easiest solution would be to roll your own dataloader based on [torch.utils.data.DataLoader](https://pytorch.org/docs/master/data.html#torch.utils.data.DataLoader) if you want a pure python version.

**Comment by Eie1** [comment]
Thanks for helping ,could I ask the following questions?

1       Now I can run model--dfnet by input :
        # spec: [B, 1, T, F, 2], F: freq_bins=481
        # erb: [B, 1, T, Fe]Fe=32
        # feac: [B, 2, T, Fc]Fc=481
        such as:
        test_spec = torch.randn(7, 1, 501, 481, 2)
        test_erb = torch.randn(7, 1, 501, 32)  # spec===torch.Size([2, 1, 501, 481])
        test_feac = torch.randn(7, 2, 501, 481)
    
        2       and get output :
                spec, m, lsnr, df_alpha = self.net(spec=test_spec, feat_erb=test_erb, feat_spec=test_feac)
        outdim of spec, m, lsnr, df_alpha are:
            #torch.Size([7, 1, 501, 481, 2])
            # torch.Size([7, 1, 501, 32]) 
            # torch.Size([7, 501, 1]) 
            # torch.Size([7, 501, 1]) 
            
        3       could I input what loss function need to make it run?
        i set self.df_loss=Loss()  , and implement the following code:

        df_alpha, multi_stage_specs = None, []
        if isinstance(other, Tensor):
            df_alpha = other
        elif isinstance(other, (list, tuple)):
            multi_stage_specs = other 
        snrs =[-5, 0, 5, 10, 20, 40]
        batch_train_df_loss = self.df_loss.forward(
                clean,
                noisy,
                enh,
                m,
                lsnr,
                df_alpha=df_alpha,
                max_freq=batch.max_freq,
                snrs=snrs,
                multi_stage_specs=multi_stage_specs,
            )

        
        4       input  of self.df_loss.forward should be :
        clean: Tensor,
        noisy: Tensor,
        enhanced: Tensor,
        mask: Tensor,
        lsnr: Tensor,
        df_alpha: Optional[Tensor],
        snrs: Tensor,
        max_freq: Optional[Tensor] = None,
        multi_stage_specs: List[Tensor] = [],
        
        as I understand , spec=enhanced, m=mask, lsnr=lsnr, df_alpha=df_alpha
        these four are stright from net output,
        and  snrs =[-5, 0, 5, 10, 20, 40],multi_stage_specs can be none
         
        could I ask that if what i input are right and  what max_freq , clean,noisy should be? 
        clean and noisy should be stft ouput of wav?
        the dim of them is  [B, 1, T, F, 2]?

Thank you very much

**Comment by Rikorose** [comment]
The loss function code is kind of hacky due to my research. You could also roll your own loss. I added some more docs for the `Loss` in `loss.py` though.

**Comment by Eie1** [comment]
Could I ask that are these not right of my four questions? I suppose the effect is related with the loss,is it right?
As for roll my own loss,could you recommend some kinds of loss?such as what loss I can use in net in loss.py or some losses I can try
Thank you for helping very much

**Comment by Rikorose** [comment]
>> Could I ask that are these not right of my four questions? I suppose the effect is related with the loss,is it right?

Sorry it is kind of hard for me to understand your questions. Please properly format your code an questions and write clear questions in more or less proper english.

>> As for roll my own loss,could you recommend some kinds of loss?such as what loss I can use in net in loss.py or some losses I can try

Why don't you just start with the default DeepFilterNet2 loss configuration?
